src/core/core.datasetController.js

```diff
@@ -268,6 +268,49 @@ helpers.extend(DatasetController.prototype, {
 		}
 	},
 
+	/*
+	 * Returns a set of predefined style properties that should be used to represent the dataset
+	 * or the data if the index is specified
+	 * @params {number} index - data index
+	 * @return {IStyleInterface} style object
+	 */
+	getStyle: function(index) {
+		var me = this;
+		var meta = me.getMeta();
+		var element, style;
+
+		if (meta.dataset && index === undefined) {
+			element = meta.dataset;
+		} else {
+			index = index || 0;
+			element = meta.data[index];
+		}
+		style = me._resolveElementOptions(element || {}, index);
+
+		if (style.fill === false || style.fill === null) {
+			style.backgroundColor = 'rgba(0,0,0,0)';
+		}
+
+		return {
+			backgroundColor: style.backgroundColor,
+			borderCapStyle: style.borderCapStyle,
+			borderDash: style.borderDash,
+			borderDashOffset: style.borderDashOffset,
+			borderJoinStyle: style.borderJoinStyle,
+			borderWidth: style.borderWidth,
+			borderColor: style.borderColor,
+			pointStyle: style.pointStyle,
+			rotation: style.rotation
+		};
+	},
+
+	/**
+	 * @private
+	 */
+	_resolveElementOptions: function() {
+		// noop
```

Comment:
It seems like this should throw an `Error` letting users know that they need to implement this method. If this method is not implemented then `getStyle` will be broken and it will be harder to track down why vs. if this threw an `Error`

Reply:
An error message was added. `_resolveElementOptions` has a lot of duplication between controllers, so probably we need further refactoring (in other PR).